x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_fetch_browser_fields_capabilities.tsx

```diff
@@ -25,56 +28,74 @@ export type UseFetchAlerts = ({ featureIds }: FetchAlertsArgs) => [boolean, Fetc
+ */
+const getDataViewStateFromIndexFields = (
+  _title: string,
+  fields: DataViewFieldBase[],
+  _includeUnmapped: boolean = false
+): BrowserFields => {
+  // Adds two dangerous casts to allow for mutations within this function
+  type DangerCastForMutation = Record<string, {}>;
+
+  return fields.reduce<BrowserFields>((browserFields, field) => {
+    // mutate browserFields
+    const category = getCategory(field.name);
+    if (browserFields[category] == null) {
+      (browserFields as DangerCastForMutation)[category] = {};
+    }
+    if (browserFields[category].fields == null) {
+      browserFields[category].fields = {};
+    }
+    (browserFields[category].fields as Record<string, BrowserField>)[field.name] =
+      field as unknown as BrowserField;
+
+    return browserFields;
+  }, {});
+};
+
 export const useFetchBrowserFieldCapabilities = ({
   featureIds,
   initialBrowserFields,
 }: FetchAlertsArgs): [boolean | undefined, BrowserFields] => {
   const {
     http,
+    data: dataService,
     notifications: { toasts },
   } = useKibana().services;
 
-  const [isLoading, setIsLoading] = useState<boolean | undefined>(undefined);
-  const [browserFields, setBrowserFields] = useState<BrowserFields>(
-    () => initialBrowserFields ?? {}
-  );
-
-  const getBrowserFieldInfo = useCallback(async () => {
-    if (!http) return Promise.resolve({});
+  const enabled = !initialBrowserFields && !featureIds.includes(INVALID_FEATURE_ID);
 
-    try {
-      return await http.get<BrowserFields>(`${BASE_RAC_ALERTS_API_PATH}/browser_fields`, {
-        query: { featureIds },
-      });
-    } catch (e) {
+  const { data: dataView, isFetching } = useQuery({
+    queryKey: ['fetchBrowserFields', featureIds],
+    queryFn: () => {
+      return loadAlertDataView({ http, dataService, featureIds });
+    },
+    onError: () => {
       toasts.addDanger(ERROR_FETCH_BROWSER_FIELDS);
-      return {};
-    }
-  }, [featureIds, http, toasts]);
-
-  useEffect(() => {
-    if (initialBrowserFields) {
-      // Event if initial browser fields is empty, assign it
-      // because client may be doing it to hide Fields Browser
-      setBrowserFields(initialBrowserFields);
-      return;
-    }
-
-    if (isLoading !== undefined || featureIds.includes(INVALID_FEATURE_ID)) {
-      return;
-    }
-
-    setIsLoading(true);
-
-    const callApi = async () => {
-      const browserFieldsInfo = await getBrowserFieldInfo();
-
-      setBrowserFields(browserFieldsInfo);
-      setIsLoading(false);
-    };
-
-    callApi();
-  }, [getBrowserFieldInfo, isLoading, featureIds, initialBrowserFields]);
+    },
+    enabled,
+    keepPreviousData: true,
+    cacheTime: 0,
+    refetchOnWindowFocus: false,
+  });
+
+  const loading = isFetching;
+
+  if (!enabled && initialBrowserFields) {
+    return [undefined, initialBrowserFields];
+  }
+
+  if (!dataView) {
+    return [undefined, {}];
+  }
+
+  const browserFields = getDataViewStateFromIndexFields(
```

Comment:
since we remove the `memoizeone` around `getDataViewStateFromIndexFields`. Should we use `useMemo` to make sure to avoid this function when we can